woocommerce/woocommerce-ios
# [Order creation/editing] Allow product quantity to be changed using keyboard

Part of: #10986 


## Description


Users have raised an issue that for some products, it's inconvenient to have to adjust the quantity with a stepper, one at a time.

This PR replaces the label showing the quantity in the edit view's stepper, with a TextField, to allow direct edits of the quantity.

## Testing instructions


1. Launch the app
2. Navigate to the Orders tap
3. Tap create, or open an existing order which is editable
4. Add a product
5. Expand the product row
6. Observe that you can edit the value either using the stepper, or by directly typing a number.

### Voiceover
1. Repeat the above test, but use VoiceOver to access the stepper.
3. Observe that you can swipe up or down to adjust the value, and the control is announced as a Text Field.
4. Double tap to edit; observe that you can edit the quantity directly, and dismiss the keyboard using the `Done` button
5. Observe that when editing the value directly, up and down swipes move the cursor position, as per normal VO behavior.

### `x` button
Test that the `-` button changes to an `x` when tapping it would remove the product from the order, even if that's because 1 or 0 have been entered with the keyboard.

### Reset when focus changed
1. Create or edit an order with multiple product rows
2. Tap the text field for the first row and make a change
3. Don't tap `Done`, but tap into the text field for another product row
4. Observe that the original row's text field resets to the correct saved state

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/2472348/6c58d0ef-bed2-4977-9b54-ab00dd27e10c



### VoiceOver 🔉
(Older screen recording, ignore design differences)
https://github.com/woocommerce/woocommerce-ios/assets/2472348/cc29b17c-2c71-48a2-a541-618c3b0fca6d


---
